elide-core/src/main/java/com/yahoo/elide/core/pagination/Pagination.java

```diff
@@ -88,23 +76,65 @@ public static Pagination parseQueryParams(final MultivaluedMap<String, String> q
 
         if (pageData.isEmpty()) {
             return DEFAULT_PAGINATION;
+        } else if (pageData.size() > 2) {
+            throw new InvalidValueException("Invalid usage of pagination parameters.");
         }
 
-        boolean isPageBased = pageData.containsKey(PaginationKey.page);
-        int limit = !pageData.containsKey(PaginationKey.limit) ? DEFAULT_PAGE_SIZE : pageData.get(PaginationKey.limit);
-        if (limit < 0) {
-            throw new InvalidValueException("page[size] and page[limit] must contain positive values.");
-        }
+        if (isPageBased(pageData)) {
+
+            // Page-based pagination strategy
+            int limit = pageData.get(PaginationKey.size);
+            if (limit > MAX_PAGE_LIMIT) {
+                throw new InvalidValueException("page[size] value must be less than or equal to " + MAX_PAGE_LIMIT);
+            } else if (limit < 0) {
+                throw new InvalidValueException("page[size] must contain a positive value.");
+            }
+
+            int pageNumber = pageData.containsKey(PaginationKey.number) ? pageData.get(PaginationKey.number) : 1;
+            if (pageNumber < 1) {
+                throw new InvalidValueException("page[number] must contain a positive value.");
+            }
+
+            int computedOffset = pageNumber > 0 ? (pageNumber - 1) * limit : 0;
+            return new Pagination(computedOffset, limit);
+
+        } else if (isOffsetBased(pageData)) {
+
+            // Offset-based pagination strategy
+            int limit = pageData.containsKey(PaginationKey.limit)
+                    ? pageData.get(PaginationKey.limit) : DEFAULT_PAGE_LIMIT;
```

Comment:
Above you check and make sure the limit <= MAX_PAGE_LIMIT (or else you throw an error).  In this clause you don't.